tests/ReferenceSqlTest.php

```diff
@@ -1032,4 +1032,144 @@ public function testHasOneReferenceType(): void
+                ['id' => 2, 'name' => 'Jane', 'country' => 'UK'],
+            ],
+            'book' => [
+                ['id' => 1, 'author_id' => 1, 'title' => 'Book 1'],
+                ['id' => 2, 'author_id' => 1, 'title' => 'Book 2'],
+            ],
+        ]);
+
+        $author = new Model($this->db, ['table' => 'author']);
+        $author->addField('name');
+        $author->addField('country');
+
+        // Closure allows aggregation on joined tables
+        $author->hasMany('Books', ['model' => static function ($persistence) {
+            $book = new Model($persistence, ['table' => 'book']);
+            $book->addField('author_id', ['type' => 'bigint']);
+            $book->addField('title');
+            $book->join('author');
+            // Json aggregation with dot notation allows skipping addField() calls for joined tables.
+            // Without dot notation, you must add all fields required in the aggregation,
+            // doing the same job twice. This is prone to errors and not low-code.
+
+            return $book;
+        }])
+            ->addField('books_with_author_json', [
+                'aggregate' => 'json',
+                'fields' => [
+                    'book_id' => 'id',       // Field from book table: new_field_name => original_field_name
+                    'book_title' => 'title', // Field from book table
+                    'author.name',           // Dot notation: reference joined field without addField(), shortened to 'name'
+                    'author.country',        // Dot notation: same, shortened to 'country'
+                ],
+                'type' => 'json',
+            ]);
+
+        $john = $author->load(1);
+        self::assertSame([
+            ['book_id' => 1, 'book_title' => 'Book 1', 'name' => 'John', 'country' => 'US'],
+            ['book_id' => 2, 'book_title' => 'Book 2', 'name' => 'John', 'country' => 'US'],
+        ], $john->get('books_with_author_json'));
+    }
+
+    public function testJsonAggregateExpressionTypes(): void
+    {
+        $this->setDb([
+            'author' => [
+                ['id' => 1, 'name' => 'John'],
+            ],
+            'book' => [
+                ['id' => 1, 'author_id' => 1, 'title' => 'Book A'],
+            ],
+        ]);
+
+        $author = new Model($this->db, ['table' => 'author']);
+        $author->addField('name');
+
+        $bookModel = static function ($persistence) {
+            $book = new Model($persistence, ['table' => 'book']);
+            $book->addField('author_id', ['type' => 'bigint']);
+            $book->addField('title');
+            $book->join('author');
+
+            return $book;
+        };
+
+        $author->hasMany('Books1', ['model' => $bookModel])
+            ->addField('books_expr', [
+                'aggregate' => 'json',
+                'fields' => ['info' => ['expr' => "CONCAT([], ' by ', [])", 'args' => ['title', 'author.name']]],
```

Comment:
better to use `Query::fxConcat()` here, might also help you fixing the CI for all DB vendors